File: leetcode/1869.scala
object Solution {
    def checkZeroOnes(s: String): Boolean = {
        val max_len = (c: Char) => (s split {if (c == '0') "1" else "0"})   
            .foldLeft(0) {_ max _.length}
        max_len('1') > max_len('0')
    }
}

File: leetcode/1869.py
class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        def longest_segment(segments: List[str], longest: int = 0):
            if not segments:
                return longest
            return longest_segment(
                segments[1:],
                max(longest, len(segments[0]))
            )

        ones, zeros = s.split('0'), s.split('1')

        return longest_segment(ones) > longest_segment(zeros)
